### PR TITLE
ACCUMULO-4798 optimized stat in ZooCache

### PR DESCRIPTION
**Pull request by keith-turner**

### CODE REVIEW

File: fate/src/main/java/org/apache/accumulo/fate/zookeeper/ZooCache.java [hunk 1]
```diff
@@ -60,14 +55,39 @@
...
+
+    public long getEphemeralOwner() {
+      return emphemeralOwner;
+    }
+
+    private void set(ZcStat cachedStat) {
+      this.emphemeralOwner = cachedStat.emphemeralOwner;
+    }
+
+    @VisibleForTesting
```
Comment (ctubbsii) [inline fate/src/main/java/org/apache/accumulo/fate/zookeeper/ZooCache.java]: That's great that this is visible only for testing, but I don't understand what the relationship is between "ephemheralOwner" and "session". Neither the method nor the test which calls it explains this.

File: fate/src/main/java/org/apache/accumulo/fate/zookeeper/ZooCache.java [hunk 2]
```diff
@@ -60,14 +55,39 @@
...
 
   private final HashMap<String,byte[]> cache;
-  private final HashMap<String,Stat> statCache;
+  private final HashMap<String,ZcStat> statCache;
   private final HashMap<String,List<String>> childrenCache;
 
   private final ZooReader zReader;
 
+  public static class ZcStat {
+    private long emphemeralOwner;
```
Comment (ctubbsii) [inline fate/src/main/java/org/apache/accumulo/fate/zookeeper/ZooCache.java]: Spelling.

File: fate/src/main/java/org/apache/accumulo/fate/zookeeper/ZooCache.java (hunk 1 above)
Comment (keith-turner) [reply fate/src/main/java/org/apache/accumulo/fate/zookeeper/ZooCache.java]: oh, alot of code uses the ephemheralOwner as a session id.  However here it should not be called session.

File: fate/src/test/java/org/apache/accumulo/fate/zookeeper/ZooCacheTest.java [hunk 3]
```diff
@@ -80,9 +80,9 @@ private void testGet(boolean fillStat) throws Exception {
     if (fillStat) {
       myStat = new ZcStat();
     }
-    long now = System.currentTimeMillis();
+    final long ephemeralOwner = 123456789l;
```
Comment (ctubbsii) [inline fate/src/test/java/org/apache/accumulo/fate/zookeeper/ZooCacheTest.java]: Using Random here might make this more clear that this number doesn't matter... we're just using it to ensure the number we put in is the same one we see later. But, a brief comment to that would also work.

**Comment by keith-turner** [comment]
Merged in eb15e45

**Pull request closed without merging**